key.ui/src/main/java/de/uka/ilkd/key/core/KeYSelectionModel.java

```diff
@@ -17,6 +17,12 @@
 public class KeYSelectionModel {
     private static final Logger LOGGER = LoggerFactory.getLogger(KeYSelectionModel.class);
 
+    // the listener to be informed before all
+    // this is the KeYMediator listener and should be removed later
+    // at the moment it is used to ensure a sane state of the mediator
+    // before other selection listeners receive change information
+    private final KeYSelectionListener primary;
```

Comment:
When exactly is "later"? What I mean is: Should we reflect its temporary status by adding a `TODO` or changing the name to, e.g., `selectionListenerToBeRemoved`?